packages/libs/eda/src/lib/map/analysis/appState.ts

```diff
@@ -342,5 +385,7 @@ export function useAppState(
     setViewport: useSetter('viewport'),
     setTimeSliderConfig: useSetter('timeSliderConfig', true),
     setStudyDetailsPanelConfig: useSetter('studyDetailsPanelConfig'),
+    setLegendPanelConfig: useSetter('legendPanelConfig'),
+    setVisualizationPanelConfigs: useSetter('visualizationPanelConfigs'),
```

Comment:
I'm hoping we don't need to add these marker-type-specific setters.
We could imagine a marker type mode that doesn't have a legend, for example. Or no visualization panels. (I am channelling Dave here... :rofl: )

I'm pretty sure we can make the updates to the legend and viz panel configs using the regular `updateConfiguration` inside the relevant XXXMarkerMapTypes. (e.g. in the `updateLegendPosition`, `updateVisualizationPosition` and `updateVisualizationDimensions` callbacks)